Ladybird/Qt/Application.cpp

```diff
@@ -43,6 +45,38 @@ bool Application::event(QEvent* event)
     return QApplication::event(event);
 }
 
+static ErrorOr<NonnullRefPtr<ImageDecoderClient::Client>> launch_new_image_decoder()
+{
+    auto paths = TRY(get_paths_for_helper_process("ImageDecoder"sv));
+    return launch_image_decoder_process(paths);
+}
+
+ErrorOr<void> Application::initialize_image_decoder()
+{
+    m_image_decoder_client = TRY(launch_new_image_decoder());
+
+    m_image_decoder_client->on_death = [this] {
+        m_image_decoder_client = nullptr;
+        if (auto err = this->initialize_image_decoder(); err.is_error()) {
+            dbgln("Failed to restart image decoder: {}", err.error());
+            return;
```

Comment:
shouldn't we crash if image decoder restart fails? continuing browsing without images seems frustrating and opening new tab/window won't help to get them back.

Reply:
hm. yeah I guess so. this should really only happen in super dire situations. normally we would expect the restart to continue and any missing images that were loaded while it was down can be recovered with a refresh.